### PR TITLE
Final test refactor

### PR DESCRIPTION
**Pull request by evan-**

### CODE REVIEW

**Comment by evan-** [comment]
Problems: Running 'rspec' on any folder but the 'factory' folder lets the program run cleanly. However, running it on the 'spec/factory/' folder, or just running 'rspec' by itself, raises the 'factory already registered: duplicate definition error'. 

The view_days_before_delete_spec raises some weird errors that I have yet to puzzle out.

File: spec/features/user_completes_todo_spec.rb [hunk 1]
```diff
@@ -1,34 +1,20 @@
 require 'rails_helper'
 
-feature 'user completes ToDo' do
+feature 'As a user, I want to mark todo items as complete and have them deleted' do
```
Comment (oliverbarnes) [inline spec/features/user_completes_todo_spec.rb]: The full user story is more useful as a comment above the feature description, as this becomes part of rspec's output (with --format documentation), and the scenario's own description get appended to it - the whole thing can be read as a sentence. I'd call this feature 'Marking todo items')

File: spec/features/user_completes_todo_spec.rb [hunk 2]
```diff
@@ -1,34 +1,20 @@
...
-    within '.user-info' do
-      click_link 'My Todos'
-    end
-
-    within '.col-md-8' do
-      expect( page ).to have_content(@todo.description)
-      click_on "Complete Todo"
-    end
-
+  scenario 'user completes a todo item' do
```
Comment (oliverbarnes) [inline spec/features/user_completes_todo_spec.rb]: here the scenario could be described as 'on the todo list'

File: spec/features/user_creates_todo_spec.rb [hunk 3]
```diff
@@ -1,16 +1,39 @@
...
+    #@user = create(:user)
+    #visit root_path
+    #find('.user-info').click_link('Sign In')
+    #fill_in 'Email', with: @user.email
+    #fill_in 'Password', with: @user.password
+    #find('.user-info').click_link('Sign In')
+    #visit new_todo_path
+  end
+
+  scenario 'creates a Todo to meet up with the team' do
```
Comment (oliverbarnes) [inline spec/features/user_creates_todo_spec.rb]: no need to have a scenario for a specific todo description. scenarios go more in the way of 'successfully', 'with a missing field', or based on interface location, like 'From the homepage'

File: spec/features/user_creates_todo_spec.rb [hunk 4]
```diff
@@ -1,16 +1,39 @@
...
     fill_in 'Description', with: 'Meet up with the team'
     click_button 'Save'
     expect( page ).to have_content('Your new TODO was saved')
     expect( page ).to have_content('Meet up with the team')
   end
 
-  scenario 'throws up error when ToDo tries to save without description' do
-    visit new_todo_path
+  scenario 'creates a Todo to research "all veal diet"' do
```
Comment (oliverbarnes) [inline spec/features/user_creates_todo_spec.rb]: this scenario is redundant with the previous one

File: spec/features/user_creates_todo_spec.rb [hunk 5]
```diff
@@ -1,16 +1,39 @@
...
+    fill_in 'Description', with: 'research "all veal diet"'
+    click_button 'Save'
+    expect( page ).to have_content('Your new TODO was saved')
+    expect( page ).to have_content('research "all veal diet"')
+  end
+
+  scenario 'raises error when ToDo tries to save without description' do
```
Comment (oliverbarnes) [inline spec/features/user_creates_todo_spec.rb]: good, this is testing an error scenario. Only it's not a raised error ('raising' is a term reserved for exceptions), and you can describe this scenario simply as 'without a description'

File: spec/features/user_sign_in_spec.rb [hunk 6]
```diff
@@ -1,29 +1,24 @@
...
-        click_link 'Sign In'
-      end
-      fill_in 'Email', with: @user.email
-      fill_in 'Password', with: @user.password
-
-      within 'form' do
-        click_button 'Sign In'
-      end
-    end
+  scenario "Successful" do
```
Comment (oliverbarnes) [inline spec/features/user_sign_in_spec.rb]: great.

File: spec/features/user_sign_up_spec.rb [hunk 7]
```diff
@@ -1,9 +1,16 @@
...
+
+  before do
+    @user = create(:user)
+  end
+
+  it "Signs user up for Blocitoff" do
+    visit new_user_registration_path
+    fill_in 'Email', with: @user.email
+    fill_in 'Password', with: @user.password
+    find('.actions').click_button('Sign Up')
```
Comment (oliverbarnes) [inline spec/features/user_sign_up_spec.rb]: needs an expectation - what does the user expect to see as feedback from the interface after signing up?

File: spec/features/user_views_profile_page_spec.rb [hunk 8]
```diff
@@ -1 +1,18 @@
...
+    @user = create(:user)
+    visit root_path
+    click_link "Sign In"
+    fill_in 'Email', with: @user.email
+    fill_in 'Password', with: @user.password
+    click_button "Sign In"
+  end
+
+  scenario 'user clicks on "My Profile" to view profile' do
+    click_link "My Profile"
```
Comment (oliverbarnes) [inline spec/features/user_views_profile_page_spec.rb]: missing an expectation

File: spec/features/view_days_before_delete_spec.rb [hunk 9]
```diff
@@ -1,15 +1,23 @@
...
+
+    visit root_path
+    click_link 'Sign In'
+    fill_in 'Email', with: @user.email
+    fill_in 'Password', with: @user.password
+    click_button 'Sign In'
+  end
 
-feature 'user *sees* how many days until a todo expires' do
   scenario 'Successfully' do
```
Comment (oliverbarnes) [inline spec/features/view_days_before_delete_spec.rb]: scenario 'after signing in' do

File: spec/features/viewing_todo_list_spec.rb [hunk 10]
```diff
@@ -1,14 +1,18 @@
...
+    fill_in 'Email', with: @user.email
+    fill_in 'Password', with: @user.password
+    click_button 'Sign In'
   end
 
-  scenario 'user logs in' do
-    visit user_todos_path
-    expect( page ).to have_content(@todo.description)
+  scenario 'User views their todo list' do
+    click_link "My Todos"
```
Comment (oliverbarnes) [inline spec/features/viewing_todo_list_spec.rb]: missing an expectation as well

File: spec/models/todo_spec.rb [hunk 11]
```diff
@@ -6,7 +6,7 @@
     describe '#days_left' do
       before do
         @todo = create(:todo)
-        @todo.created_at = (Time.now - 283000)
+       @todo.created_at = (Time.now - 283000)
```
Comment (oliverbarnes) [inline spec/models/todo_spec.rb]: you can use 3.days.ago

File: spec/models/user_spec.rb [hunk 12]
```diff
@@ -1,11 +1,6 @@
 require 'rails_helper'
 
 describe User do
```
Comment (oliverbarnes) [inline spec/models/user_spec.rb]: empty spec, can be left out of the pull request

**Pull request merged**